ansible_files/aws_ec2_region_cleanup.yml

```diff
@@ -0,0 +1,148 @@
+          {%-   set uptime_days = uptime_total_seconds // 86400 -%}
+          {%-   set remainder_seconds_after_days = uptime_total_seconds % 86400 -%}
+          {%-   set uptime_hours = remainder_seconds_after_days // 3600 -%}
+          {%-   set remainder_seconds_after_hours = remainder_seconds_after_days % 3600 -%}
+          {%-   set uptime_minutes = remainder_seconds_after_hours // 60 -%}
+          {%-   set uptime_str_parts = [] -%}
+          {%-   if uptime_days > 0 %}{%- set _ = uptime_str_parts.append(uptime_days ~ 'd') -%}{%- endif -%}
+          {%-   if uptime_hours > 0 %}{%- set _ = uptime_str_parts.append(uptime_hours ~ 'h') -%}{%- endif -%}
+          {%-   if uptime_minutes > 0 %}{%- set _ = uptime_str_parts.append(uptime_minutes ~ 'm') -%}{%- endif -%}
+          {%-   if uptime_str_parts | length == 0 %}{%- set _ = uptime_str_parts.append('<1m') -%}{%- endif -%}
+          {%-   set uptime_str = uptime_str_parts | join(' ') -%}
+          {%-   set inst_name = inst.tags.Name | default('N/A') -%}
+          {%-   set display_launch_time = inst.launch_time_str | regex_replace('\\+00:00$', 'Z') | regex_replace('T', ' ') -%}
+          {%-   set ns.report_text = ns.report_text ~ "  - Instance ID: " ~ inst.id ~ "\n" -%}
+          {%-   set ns.report_text = ns.report_text ~ "    Name: " ~ inst_name ~ "\n" -%}
+          {%-   set ns.report_text = ns.report_text ~ "    Launch Time: " ~ display_launch_time ~ "\n" -%}
+          {%-   set ns.report_text = ns.report_text ~ "    Uptime: " ~ uptime_str ~ " (" ~ uptime_total_seconds ~ " sec)" -%}
+          {%-   if not loop.last -%}
+          {%-     set ns.report_text = ns.report_text ~ "\n\n" -%} {# Explicitly add two newlines between instances #}
+          {%-   else -%}
+          {%-     set ns.report_text = ns.report_text ~ "\n" -%} {# Single newline after the very last instance's details #}
+          {%-   endif -%}
+          {%- endfor -%}
+          {{ ns.report_text }}
+      when: leaked_instances is defined and leaked_instances | length > 0
+
+    - name: Compose Slack message content
+      ansible.builtin.set_fact:
+        slack_message_text: |
+          {% if slack_mention | trim != "" %}{{ slack_mention }} {% endif %} :aws:  *EC2 Leaked Instance Report & Action*
+          *Region:* `{{ aws_region }}`
+          *Uptime Threshold:* Over {{ (leak_threshold_seconds | int) // 3600 }}h {{ ((leak_threshold_seconds | int) % 3600) // 60 }}m ({{ leak_threshold_seconds | int }} seconds)
+          *Termination Enabled:* `{{ terminate_leaked_instances_flag }}`
+          ---
+          {% if leaked_instances is defined and leaked_instances | length > 0 %}
+          Found *{{ leaked_instances | length }}* leaked instance(s):
+          ```
+          {{ report_lines_for_slack }}```
+          {% else %}
+          No EC2 instances found exceeding the leak uptime threshold in region `{{ aws_region }}`.
```

Comment:
do we really want to send a message if there is nothing to do? I think it would reduce noise on the channel

Reply:
Cancelled it